src/gallery.tsx

```diff
@@ -319,17 +325,19 @@ export const Gallery: FC<GalleryProps> = ({
       const { id } = data
       items.current.set(ref, data)
 
-      if (!openWhenReadyPid.current) return
-
-      if (id === openWhenReadyPid.current) {
-        open(ref)
-        openWhenReadyPid.current = null
-      } else if (!id) {
-        const index = parseInt(openWhenReadyPid.current, 10) - 1
-        const refToOpen = Array.from(items.current.keys())[index]
-        if (refToOpen) {
-          open(refToOpen)
+      if (openWhenReadyPid.current !== null) {
+        if (id === openWhenReadyPid.current) {
+          // user provided `id` prop of Item component
+          open(ref)
           openWhenReadyPid.current = null
+        } else if (!id) {
+          // in this case we using index of item as PID
+          const index = parseInt(openWhenReadyPid.current, 10) - 1
+          const refToOpen = Array.from(items.current.keys())[index]
+          if (refToOpen) {
+            open(refToOpen)
+            openWhenReadyPid.current = null
+          }
```

Comment:
mb we should remove such deep ``if`` nesting?

Reply:
changed in [d98b36b](https://github.com/dromru/react-photoswipe-gallery/pull/592/commits/d98b36be61993dda6659a88db0186a85bb5a0624)